File: example/language/hello.js
#!/usr/bin/env node

/**
 * JavaScript 示例文件
 * 展示 ES6+ 特性
 */

// 导入模块
const fs = require('fs');
const path = require('path');

// 全局常量
const PI = 3.14159;
const APP_NAME = "JavaScript Example";

// 类定义
class Animal {
    constructor(name, age) {
        this.name = name;
        this.age = age;
    }

    speak() {
        console.log(`${this.name} makes a sound`);
    }

    getInfo() {
        return `${this.name} is ${this.age} years old`;
    }

    static createDog(name) {
        return new Dog(name, 0);
    }
}

class Dog extends Animal {
    constructor(name, age) {
        super(name, age);
    }

    speak() {
        console.log(`${this.name} barks!`);
    }

    fetch(item) {
        console.log(`${this.name} fetches the ${item}`);
    }
}

// 函数声明
function greet(name) {
    return `Hello, ${name}!`;
}

// 箭头函数
const calculateArea = (radius) => {
    return PI * radius * radius;
};

const multiply = (a, b) => a * b;

// 异步函数
async function fetchData(url) {
    try {
        const response = await fetch(url);
        const data = await response.json();
        return data;
    } catch (error) {
        console.error('Error fetching data:', error);
        throw error;
    }
}

// 生成器函数
function* fibonacciGenerator() {
    let a = 0, b = 1;
    while (true) {
        yield a;
        [a, b] = [b, a + b];
    }
}

// 模板字符串和解构赋值
function processUser({name, age, email = 'unknown@example.com'}) {
    const message = `
        User Information:
        Name: ${name}
        Age: ${age}
        Email: ${email}
        Greeting: ${greet(name)}
    `;
    return message;
}

// 数组方法
const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];

const doubled = numbers.map(num => num * 2);
const evens = numbers.filter(num => num % 2 === 0);
const sum = numbers.reduce((acc, curr) => acc + curr, 0);

// 对象字面量增强
const createPerson = (name, age) => ({
    name,
    age,
    greet() {
        return `Hi, I'm ${this.name}`;
    },
    get isAdult() {
        return this.age >= 18;
    },
    set updateAge(newAge) {
        this.age = newAge;
    }
});

// Promise 示例
const delay = (ms) => new Promise(resolve => setTimeout(resolve, ms));

async function main() {
    console.log(greet("JavaScript"));

    // 变量声明
    let mutableVar = "I can change";
    const constantVar = "I cannot change";

    // 数据类型
    const string = "Hello World";
    const number = 42;
    const boolean = true;
    const array = [1, 2, 3];
    const object = {key: "value"};

    // 条件语句
    if (number > 40) {
        console.log("Number is greater than 40");
    } else if (number === 42) {
        console.log("The answer to everything");
    } else {
        console.log("Number is small");
    }

    // 循环
    for (let i = 0; i < 3; i++) {
        console.log(`Iteration ${i}`);
    }

    // for...of 循环
    for (const num of numbers) {
        if (num > 5) break;
        console.log(`Number: ${num}`);
    }

    // 类使用
    const dog = new Dog("Buddy", 3);
    dog.speak();
    dog.fetch("ball");
    console.log(dog.getInfo());

    // 工厂方法
    const anotherDog = Animal.createDog("Max");
    anotherDog.speak();

    // 函数调用
    console.log(`Area of circle with radius 5: ${calculateArea(5)}`);
    console.log(`5 * 3 = ${multiply(5, 3)}`);

    // 数组操作
    console.log(`Doubled numbers: ${doubled.join(', ')}`);
    console.log(`Even numbers: ${evens.join(', ')}`);
    console.log(`Sum: ${sum}`);

    // 对象使用
    const person = createPerson("Alice", 30);
    console.log(person.greet());
    console.log(`Is adult: ${person.isAdult}`);
    person.updateAge = 25;
    console.log(`New age: ${person.age}`);

    // 用户处理
    const user = {name: "Bob", age: 25};
    console.log(processUser(user));

    // 生成器
    const fibGen = fibonacciGenerator();
    console.log("Fibonacci numbers:");
    for (let i = 0; i < 10; i++) {
        console.log(fibGen.next().value);
    }

    // 异步操作
    console.log("Starting async operation...");
    await delay(1000);
    console.log("Async operation completed");

    // 可选链和空值合并
    const obj = {a: {b: {c: "value"}}};
    const safeAccess = obj?.a?.b?.c ?? "default";
    console.log(`Safe access result: ${safeAccess}`);

    // BigInt
    const bigNum = 1234567890123456789012345678901234567890n;
    console.log(`BigInt: ${bigNum}`);

    // Symbol
    const sym = Symbol("unique");
    console.log(`Symbol: ${sym.toString()}`);

    console.log("JavaScript example completed!");
}

// 导出模块（如果作为模块使用）
module.exports = {
    Animal,
    Dog,
    greet,
    calculateArea,
    fibonacciGenerator
};

// 如果直接运行此文件
if (require.main === module) {
    main().catch(console.error);
}

File: example/language/hello.py
#!/usr/bin/env python3
"""
Python示例文件 - 展示各种语法高亮特性
"""

import sys
import os
from typing import List, Dict, Optional
from dataclasses import dataclass

# 全局变量
GLOBAL_VAR = "Hello World"

@dataclass
class Person:
    """数据类示例"""
    name: str
    age: int
    email: Optional[str] = None

    def greet(self) -> str:
        return f"Hello, my name is {self.name}"

class Calculator:
    """计算器类"""

    def __init__(self, initial_value: float = 0.0):
        self.value = initial_value

    def add(self, number: float) -> float:
        self.value += number
        return self.value

    def multiply(self, number: float) -> float:
        self.value *= number
        return self.value

    @staticmethod
    def factorial(n: int) -> int:
        if n <= 1:
            return 1
        return n * Calculator.factorial(n - 1)

def fibonacci(n: int) -> int:
    """递归计算斐波那契数"""
    if n <= 1:
        return n
    return fibonacci(n-1) + fibonacci(n-2)

def main():
    """主函数"""
    print("Hello, Python World!")

    # 变量和数据类型
    integer_var = 42
    float_var = 3.14159
    string_var = "Python is awesome"
    boolean_var = True
    list_var = [1, 2, 3, 4, 5]
    dict_var = {"key": "value", "number": 123}
    tuple_var = (1, 2, "three")

    # 字符串操作
    formatted_string = f"Integer: {integer_var}, Float: {float_var:.2f}"
    print(formatted_string)

    # 条件语句
    if integer_var > 40:
        print("Integer is greater than 40")
    elif integer_var == 42:
        print("The answer to everything")
    else:
        print("Integer is small")

    # 循环
    print("List iteration:")
    for item in list_var:
        print(f"  Item: {item}")

    print("Range iteration:")
    for i in range(3):
        print(f"  Iteration {i}")

    # 列表推导式
    squares = [x**2 for x in range(10) if x % 2 == 0]
    print(f"Squares: {squares}")

    # 字典推导式
    square_dict = {x: x**2 for x in range(5)}
    print(f"Square dict: {square_dict}")

    # 异常处理
    try:
        result = 10 / 0
    except ZeroDivisionError as e:
        print(f"Caught exception: {e}")
    finally:
        print("This always executes")

    # 类使用
    calc = Calculator(10)
    calc.add(5)
    calc.multiply(2)
    print(f"Calculator result: {calc.value}")

    person = Person("Alice", 30, "alice@example.com")
    print(person.greet())

    # 函数调用
    fib_10 = fibonacci(10)
    fact_5 = Calculator.factorial(5)
    print(f"Fibonacci(10) = {fib_10}")
    print(f"Factorial(5) = {fact_5}")

    # 装饰器示例（简单版本）
    def simple_decorator(func):
        def wrapper(*args, **kwargs):
            print(f"Calling {func.__name__}")
            return func(*args, **kwargs)
        return wrapper

    @simple_decorator
    def decorated_function():
        return "This function is decorated"

    print(decorated_function())

if __name__ == "__main__":
    main()
